File: ruby/lib/exceptions/rcon_ban_exception.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2009-2011, Sebastian Staudt

require 'exceptions/steam_condenser_exception'

# This exception class indicates that the IP address your accessing the game
# server from has been banned by the server
#
# You or the server operator will have to unban your IP address on the server.
#
# @author Sebastian Staudt
# @see GameServer#rcon_auth
class RCONBanException < SteamCondenserException

  # Creates a new `RCONBanException` instance
  def initialize
    super 'You have been banned from this server.'
  end

end

File: php/lib/exceptions/RCONBanException.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2009, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Exceptions
 */

require_once 'exceptions/SteamCondenserException.php';

/**
 * @package Steam Condenser (PHP)
 * @subpackage Exceptions
 */
class RCONBanException extends SteamCondenserException
{
	public function __construct()
	{
		parent::__construct('You have been banned from this server.');
	}
}
?>
